src/GloryDuckTables2019Lkl.cc

```diff
@@ -0,0 +1,403 @@
+{
+  if(InterpretInputString(inputString))
+    cout << "GloryDuckTables2019Lkl::GloryDuckTables2019Lkl Warning: there were problems interpreting the input string" << endl;
+}
+//////////////////////////////////////////////////////////////////
+//
+// Interpret the input configuration string
+// The string has been passed to Lkl::InterpretInputString in constructor
+// here it is searched for the following options:
+//
+// inputfile=<val>:     name of the input file, which contains the lkl vs <sv> tables
+//
+Int_t GloryDuckTables2019Lkl::InterpretInputString(TString inputString)
+{
+  // file name and path default values
+  TString inputfileName = " (No file has been specified) ";
+  TString path          = "";
+
+  // Prepeare to interpret inputString
+  TPMERegexp re("\\s+");
+  TPMERegexp fldre("=");
+  
+  // split the inputString into the different fields, and check the option and values specified in each of them
+  UInt_t nfields = re.Split(inputString);
+  for(UInt_t ifield=0;ifield<nfields;ifield++)
+    {
+      fldre.Split(re[ifield]);
+      TString optname = fldre[0];
+      if(optname.CompareTo("inputfile",TString::kIgnoreCase)==0)
+	inputfileName=fldre[1];
+      else if(optname.CompareTo("path",TString::kIgnoreCase)==0)
+	path=fldre[1];    
+    }
+
+  fSampleArray = new TObjArray();
+
+  // open and read input file
+  return ReadGloryDuckInputData(path+"/"+inputfileName);
+}
+
+////////////////////////////////////////////////////////////////
+// 
+// Destructor
+//
+GloryDuckTables2019Lkl::~GloryDuckTables2019Lkl()
+{
+  if(fSampleArray) delete fSampleArray;
+  if(fsvMin)       delete [] fsvMin;
+  if(fsvMax)       delete [] fsvMax;
+  if(fMass)        delete [] fMass;
+}
+
+////////////////////////////////////////////////////////////////
+//
+// Link fMinuit to the static global variable minuit (otherwise 
+// it won't work because of minuit subtleties)
+//
+void GloryDuckTables2019Lkl::SetMinuitLink()
+{
+  minuit = fMinuit; // copy the pointer so that it can be used by FCN 
+}
+
+////////////////////////////////////////////////////////////////
+//
+// Pass the -2logL function to fMinuit,
+// initialize the parameters (free+nuisance),
+//
+void GloryDuckTables2019Lkl::SetFunctionAndPars(Double_t ginit)
+{
+  fMinuit->SetFCN(gloryDuckTables2019Lkl);
+  fMinuit->SetName(Form("%s_Minuit",GetName()));
+
+  // Initialize minuit for added objects
+  TObjArrayIter* iter    = (TObjArrayIter*) GetSampleArray()->MakeIterator();
+  Lkl* sample;
+  while((sample=(Lkl*)iter->Next()))
+    {
+      sample->InitMinuit();
+      sample->SetMinuitLink();
+    }
+  delete iter;
+
+  Double_t pStart[gNPars] = {(ginit==0? gRefsv :ginit)};
+  Double_t pDelta[gNPars] = {1};
+
+  SetParameters(gParName,pStart,pDelta);
+}  
+
+////////////////////////////////////////////////////////////////
+//
+// Check that everything is correct
+//
+// Return 0 in case of success, 1 otherwise
+//
+Int_t GloryDuckTables2019Lkl::MakeChecks()
```

Comment:
Add check that the different data members (fNMasses, fsv,fMass,...) exist and have sensible values (e.g. not NULL)